### PR TITLE
fix(coda): upgrade SDK to v2.0.0, add unit and integration tests

### LINKED ISSUE #305: fix(coda): upgrade SDK to v2.0.0, add unit and integration tests
Upgrade autohive-integrations-sdk from ~=1.0.2 to ~=2.0.0, migrate all action handlers to SDK 2.0 patterns (FetchResponse, ActionResult, ActionError), fix optional/required param access patterns, and add comprehensive unit/integration tests.

### PR DESCRIPTION
**Pull request by Shubhank-Jonnada**
Closes #305

## Summary

- Upgraded `autohive-integrations-sdk` from `~=1.0.2` to `~=2.0.0`
- Migrated all action handlers to SDK 2.0 patterns: `FetchResponse` object (`response.data`, `response.status`), `ActionResult`, and `ActionError`
- Fixed optional vs required param access: required params use `inputs["param"]`, optional use `inputs.get("param")`
- Bumped integration version from `1.0.0` to `2.0.0`
- Removed redundant `result: True` field from all `ActionResult` returns and output schemas
- Added comprehensive unit and integration test suites covering all 20 actions

## Test Results

<details>
<summary>Unit Tests — 47/47 passed</summary>

| Test Class | Tests | Status |
|---|---|---|
| `TestListDocs` | 3 | ✅ PASSED |
| `TestGetDoc` | 2 | ✅ PASSED |
| `TestCreateDoc` | 3 | ✅ PASSED |
| `TestListPages` | 3 | ✅ PASSED |
| `TestGetPage` | 2 | ✅ PASSED |
| `TestListTables` | 2 | ✅ PASSED |
| `TestGetTable` | 2 | ✅ PASSED |
| `TestListColumns` | 2 | ✅ PASSED |
| `TestGetColumn` | 2 | ✅ PASSED |
| `TestListRows` | 3 | ✅ PASSED |
| `TestGetRow` | 2 | ✅ PASSED |
| `TestUpsertRows` | 2 | ✅ PASSED |
| `TestDeleteRow` | 2 | ✅ PASSED |
| `TestDeleteRows` | 2 | ✅ PASSED |
| `TestUpdateDoc` | 2 | ✅ PASSED |
| `TestDeleteDoc` | 2 | ✅ PASSED |
| `TestCreatePage` | 3 | ✅ PASSED |
| `TestUpdatePage` | 2 | ✅ PASSED |
| `TestDeletePage` | 2 | ✅ PASSED |
| `TestUpdateRow` | 2 | ✅ PASSED |
| **Total** | **47** | ✅ **ALL PASSED** |

</details>

<details>
<summary>Integration Tests (live Coda API) — 5 passed, 19 skipped, 0 failed</summary>

24 tests covering all 20 actions. Skipped tests are chained tests that require pre-existing docs/tables/rows in the account — they skip gracefully when the account is empty, not a failure.

| Test | Action | Status |
|---|---|---|
| `test_returns_docs_list` | `list_docs` | ✅ PASSED |
| `test_filter_by_query` | `list_docs` with `query` | ✅ PASSED |
| `test_is_owner_filter` | `list_docs` with `is_owner=True` | ✅ PASSED |
| `test_limit_respected` | `list_docs` with `limit` | ✅ PASSED |
| `TestDocLifecycle` | `create_doc` → `update_doc` → `delete_doc` | ✅ PASSED |
| `TestGetDoc` (2 tests) | `get_doc` | ⏭ SKIPPED (no docs in account) |
| `TestListPages` (2 tests) | `list_pages` | ⏭ SKIPPED (no docs in account) |
| `TestGetPage` (2 tests) | `get_page` | ⏭ SKIPPED (no docs in account) |
| `TestListTables` (2 tests) | `list_tables` | ⏭ SKIPPED (no docs in account) |
| `TestGetTable` (2 tests) | `get_table` | ⏭ SKIPPED (no docs in account) |
| `TestListColumns` (2 tests) | `list_columns` | ⏭ SKIPPED (no docs in account) |
| `TestGetColumn` | `get_column` | ⏭ SKIPPED (no docs in account) |
| `TestListRows` (2 tests) | `list_rows` | ⏭ SKIPPED (no docs in account) |
| `TestGetRow` | `get_row` | ⏭ SKIPPED (no docs in account) |
| `TestPageLifecycle` | `create_page` → `update_page` → `delete_page` | ⏭ SKIPPED (no docs in account) |
| `TestRowLifecycle` | `upsert_rows` → `update_row` → `delete_row` | ⏭ SKIPPED (no docs in account) |
| `TestDeleteRows` | `delete_rows` | ⏭ SKIPPED (no docs in account) |

</details>

<details>
<summary>Validation</summary>

| Check | Result |
|---|---|
| `validate_integration.py` | ✅ PASSED (0 errors, 0 warnings) |
| `check_code.py` | ✅ PASSED (0 warnings) |
| `ruff check` | ✅ clean |

</details>

## How to run

```bash
# Unit tests
python -m pytest coda/tests/test_coda_unit.py -m unit -v

# Integration tests (requires API key)
CODA_API_KEY=<your-key> python -m pytest coda/tests/test_coda_integration.py -m integration -v

# Destructive tests only
CODA_API_KEY=<your-key> python -m pytest coda/tests/test_coda_integration.py -m "integration and destructive" -v
```

### CODE REVIEW

File: coda/coda.py [hunk 1]
```diff
@@ -21,8 +21,7 @@ def get_auth_headers(context: ExecutionContext) -> Dict[str, str]:
     Returns:
         Dictionary with Authorization and Content-Type headers
     """
-    credentials = context.auth.get("credentials", {})
-    api_token = credentials.get("api_token", "")
+    api_token = context.auth.get("api_token", "")
```
[the remark this answers is not in the document]
Comment (TheRealAgentK) [reply coda/coda.py]: I think this needs a check @Shubhank-Jonnada

[the remark this answers is not in the document]
Comment (Shubhank-Jonnada) [reply coda/coda.py]: fixed this here: https://github.com/Autohive-AI/autohive-integrations/pull/307/commits/8289ca1ba52b0ed80b2c76c514a79a02a3f8455f

File: .env.example [hunk 2]
```diff
@@ -59,3 +59,9 @@
 
 # -- Xero --
 # (uses platform OAuth — tokens are short-lived, typically not set here)
+
+# -- Coda --
+# CODA_API_KEY=
+
+# -- ElevenLabs --
+# ELEVENLABS_API_KEY=
```
Comment (TheRealAgentK) [inline .env.example]: This should not be part of this PR - CODA only.

**Comment by TheRealAgentK** [comment]
PR description is stale — it says "Bumped integration version from 1.0.0 to 1.1.0", but the final state is 2.0.0 (which is correct per the SDK v2 upgrade convention). Update the description to match.

**Comment by TheRealAgentK** [comment]
Hey @Shubhank-Jonnada — really nice work on this upgrade! 🎉 The SDK 2.0 migration looks clean and the validation/lint/tests all pass locally for me.

To round out the unit test coverage I went ahead and pushed [3c23461](https://github.com/Autohive-AI/autohive-integrations/commit/3c23461) which adds happy-path + error tests for the 6 actions that didn't have unit coverage yet:

- `update_doc`
- `delete_doc`
- `create_page` (incl. a test for the optional content/subtitle/icon/parent fields)
- `update_page`
- `delete_page`
- `update_row`

Test count goes from **34 → 47**, all green

**Comment by TheRealAgentK** [comment]
One more comment on the `ActionResult` and `ActionError` setup:

`"result": True` is now dead weight, every action still returns `"result": True` in its data and declares it required in the output schema. 

Since failure now goes through `ActionError`, this boolean is always True. The upgrade-skill explicitly calls this out: 

> "Remove the 'error' property (and any 'result: bool' property used only for error signalling) from each action's output schema." 

Maybe drop it from both the code and schemas, or leave it documented as a legacy field, but the current state is somewhat redundant.

File: .env.example (hunk 2 above)
Comment (Shubhank-Jonnada) [reply .env.example]: Removed, was accidentally left in.

**Comment by Shubhank-Jonnada** [comment]
@TheRealAgentK 
Fixed all the feedback from the review:

- Removed the ElevenLabs entry from .env.example, was accidentally left in
- Dropped `"result": True` from all ActionResult returns in coda.py and removed the result property + required entry from every output schema in config.json
- Updated the PR description to reflect the correct version (2.0.0) and the result field cleanup

**Comment by TheRealAgentK** [comment]
I get this when I'm trying to run integration tests:

```
coda/tests/test_coda_integration.py:70: in test_is_owner_filter
    assert data["result"] is True
           ^^^^^^^^^^^^^^
E   KeyError: 'result'
```

**Review by TheRealAgentK — CHANGES_REQUESTED** [review]

**Comment by Shubhank-Jonnada** [comment]
Hey @TheRealAgentK, expanded the integration test suite to cover all 20 actions (was 3 before). Also fixed the stale `data["result"]` assertion you caught.

<details>
<summary>Integration test results  5 passed, 19 skipped, 0 failed</summary>

24 tests across all 20 actions. Skips are chained tests that need pre-existing docs/tables/rows  the test account is empty so they skip gracefully, not a failure. Run against an account with actual content and all 24 will pass.

| Test | Action | Status |
|---|---|---|
| `test_returns_docs_list` | `list_docs` | ✅ PASSED |
| `test_filter_by_query` | `list_docs` with `query` | ✅ PASSED |
| `test_is_owner_filter` | `list_docs` with `is_owner=True` | ✅ PASSED |
| `test_limit_respected` | `list_docs` with `limit` | ✅ PASSED |
| `TestDocLifecycle` | `create_doc` → `update_doc` → `delete_doc` | ✅ PASSED |
| `TestGetDoc` (2 tests) | `get_doc` | ⏭ SKIPPED (no docs in account) |
| `TestListPages` (2 tests) | `list_pages` | ⏭ SKIPPED (no docs in account) |
| `TestGetPage` (2 tests) | `get_page` | ⏭ SKIPPED (no docs in account) |
| `TestListTables` (2 tests) | `list_tables` | ⏭ SKIPPED (no docs in account) |
| `TestGetTable` (2 tests) | `get_table` | ⏭ SKIPPED (no docs in account) |
| `TestListColumns` (2 tests) | `list_columns` | ⏭ SKIPPED (no docs in account) |
| `TestGetColumn` | `get_column` | ⏭ SKIPPED (no docs in account) |
| `TestListRows` (2 tests) | `list_rows` | ⏭ SKIPPED (no docs in account) |
| `TestGetRow` | `get_row` | ⏭ SKIPPED (no docs in account) |
| `TestPageLifecycle` | `create_page` → `update_page` → `delete_page` | ⏭ SKIPPED (no docs in account) |
| `TestRowLifecycle` | `upsert_rows` → `update_row` → `delete_row` | ⏭ SKIPPED (no docs in account) |
| `TestDeleteRows` | `delete_rows` | ⏭ SKIPPED (no docs in account) |

</details>

**Comment by TheRealAgentK** [comment]
:white_check_mark: All 4 list_docs tests + TestDocLifecycle::test_full_lifecycle (real create→update→delete round-trip on Coda)
**19 skipped** — they need pre-existing object IDs from env vars (TEST_DOC_ID, TEST_PAGE_ID, TEST_TABLE_ID, TEST_COLUMN_ID, TEST_ROW_ID)
:x: 0 failures

The 19 skipped tests are a known issue. @Shubhank-Jonnada, up to you if want to create a ticket to deal with them later and get the core change out (happy to approve on that basis) or if you want to try to improve these tests.

**Comment by TheRealAgentK** [comment]
## Note on integration test skips

Ran the live integration suite locally — **5 passed, 19 skipped, 0 failed**. The skips are all gated on the user's Coda account already containing docs / tables / rows / columns; an empty account skips them gracefully.

We considered lifting `TestDocLifecycle`'s `create_doc → delete_doc` into a module-scoped `seeded_doc` fixture so the rest of the suite could run end-to-end. The catch is that Coda's REST API **does not expose `create_table` / `create_column`** — tables can only be materialized via the Coda UI or packs. So a seeded doc would unskip only the doc-level and page-level tests (a fresh doc has a default canvas page); everything table/column/row would still skip.

To get the remaining ~9 tests running you'd need to either:
1. Create a "test fixtures" doc once in the Coda account, then set `CODA_TEST_DOC_ID` / `CODA_TEST_TABLE_ID` env vars and have the tests read those instead of listing the first available doc, **or**
2. Accept that they only execute when the test account happens to have a populated doc.

Not blocking — flagging so the next person isn't surprised by the skip count.

**Review by TheRealAgentK — DISMISSED** [review]
Approved, but needs a neutral review because both @Shubhank-Jonnada and I worked on this quite a bit now :)

**Review by TheRealAgentK — DISMISSED** [review]

**Review by ProRedCat — DISMISSED** [review]

**Review by NinosMan — DISMISSED** [review]

**Comment by Shubhank-Jonnada** [comment]
## Live Integration Test Results

All 28 tests passing against the real Coda API.

<details>
<summary>Full results (28 passed, 0 failed)</summary>

| Category | Action | Result |
|---|---|---|
| Docs | list_docs | PASSED |
| Docs | get_doc | PASSED |
| Docs | update_doc | PASSED |
| Docs | create_doc | PASSED |
| Docs | delete_doc | PASSED |
| Pages | list_pages | PASSED |
| Pages | get_page | PASSED |
| Pages | update_page | PASSED |
| Pages | create_page | PASSED |
| Pages | delete_page | PASSED |
| Tables | list_tables | PASSED |
| Tables | get_table | PASSED |
| Columns | list_columns | PASSED |
| Columns | get_column | PASSED |
| Rows | list_rows | PASSED |
| Rows | get_row | PASSED |
| Rows | upsert_rows | PASSED |
| Rows | update_row | PASSED |
| Rows | delete_row | PASSED |
| Rows | delete_rows | PASSED |

</details>

**Notes:**
- Session fixtures create and tear down a fresh test doc + page, so tests are fully idempotent
- Table/row tests require a doc with at least one table in the account (Coda API does not support creating tables programmatically)
- Run with: `CODA_API_KEY=<token> pytest coda/tests/test_coda_integration.py -m integration -o "addopts=--import-mode=importlib --tb=short"`

**Review by TheRealAgentK — APPROVED** [review]
